experimental/Schemes/BlowupMorphism.jl

```diff
@@ -0,0 +1,255 @@
+                             }
+  projective_bundle::CoveredProjectiveScheme 
+  codomain::CodomainType   # i.A. ein CoveredScheme
+  center::IdealSheaf      # on codomain
+  projection::AbsCoveredSchemeMorphism
+  domain::AbsCoveredScheme # i.A. ein CoveredScheme
+  exceptional_divisor::EffectiveCartierDivisor
+
+  function BlowupMorphism(
+      IP::CoveredProjectiveScheme,
+      I::IdealSheaf
+    )
+    X = base_scheme(IP)
+    X === scheme(I) || error("ideal sheaf not compatible with blown up variety")
+    return new{typeof(X)}(IP, X, I)
+  end
+end
+
+function domain(p::BlowupMorphism)
+  if !isdefined(p, :domain)
+    p.domain = covered_scheme(p.projective_bundle)
+  end
+  return p.domain
+end
+
+codomain(p::BlowupMorphism) = p.codomain
+center(p::BlowupMorphism) = p.center
+
+function projection(p::BlowupMorphism)
+  if !isdefined(p, :projection)
+    p.projection = covered_projection_to_base(p.projective_bundle)
+  end
+  return p.projection
+end
+
+# TODO: Find better name!
+covered_projective_scheme(p::BlowupMorphism) = p.projective_bundle
+
+@Markdown.doc """
+    exceptional_divisor(p::BlowupMorphism)
+
+For a `BlowupMorphism` ``p : Y → X`` coming from the blowup of an 
+`IdealSheaf` ``ℐ`` on X, return the `EffectiveCartierDivisor` ``E`` 
+on ``Y`` associated to the (relative) tautological bundle ``𝒪(1)``. 
+
+On a pair of charts ``V → U`` of the `covered_scheme` of the 
+`projection` of ``p`` this returns the pullback of the `i`-th 
+generator of ``ℐ(U)`` when ``V`` is the `i-1`-st canonical chart 
+of the local blowup over ``U``.
+"""
+function exceptional_divisor(p::BlowupMorphism)
+  if !isdefined(p, :exceptional_divisor)
+    error("exceptional divisor needs to be cached during construction")
+    Y = domain(p)
+    pr = projection(p)
+    pr_cov = covering_morphism(pr)
+    IC = center(p)
+    ideals_on_patches = IdDict{AbsSpec, Ideal}()
+    mult_flag = false # To compute the multiplicity only once 
+    multiplicity = one(ZZ)
+    for U in domain(pr_cov)
+      pr_U = pr_cov[U]
+      V = codomain(pr_U)
+      J = ideal(OO(U), pullback(pr_U).(gens(IC(V))))
+      J_rad = radical(J)
+      # TODO: Pick the "simplest" chart for the following computation.
+      # We could also first use elimpart for simplification.
+      if !mult_flag && !(J == J_rad)
+        R = base_ring(OO(U))
+        S = MPolyComplementOfPrimeIdeal(saturated_ideal(J_rad))
+        R_loc, loc_map = localization(R, S)
+        F = FreeMod(R_loc, 1)
+        JF, _ = sub(F, [g*F[1] for g in gens(saturated_ideal(J))])
+        M, _ = quo(F, JF)
+        multiplicity = length(M)
+        mult_flag = true
+      end
+      ideals_on_patches[U] = J_rad
+    end
+    IE = IdealSheaf(Y, ideals_on_patches)
+    p.exceptional_divisor = WeilDivisor(Y, ZZ, IdDict{IdealSheaf, elem_type(ZZ)}(IE => multiplicity))
+  end
+  return p.exceptional_divisor
+end
+
+### Auxiliary function which were not there; to be moved
```

Comment:
Depending on where they belong, this might be a delicate thing to do. Other people working there might complain about merge conflicts. But I'll look into that.